### PR TITLE
Evaluator module

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Move the evaluator logic to its own module.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1036)

### CODE REVIEW

File: evaluator/src/test/kotlin/EvaluatorTest.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.evaluator
+
+import com.here.ort.model.OrtResult
+import com.here.ort.model.Repository
+import com.here.ort.model.VcsInfo
+import com.here.ort.model.config.RepositoryConfiguration
+import io.kotlintest.matchers.containExactly
```
Comment (sschuberth) [inline evaluator/src/test/kotlin/EvaluatorTest.kt]: Nit: Missing blank line before this one.

File: evaluator/src/test/kotlin/EvaluatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.WordSpec
+
+class EvaluatorTest : WordSpec() {
+    init {
+        val ortResult = OrtResult(Repository(VcsInfo.EMPTY, VcsInfo.EMPTY, RepositoryConfiguration()))
+
+        "checkSyntax" should {
+            "succeed if the script can be compiled" {
+                val result = Evaluator().checkSyntax(ortResult, "")
```
Comment (sschuberth) [inline evaluator/src/test/kotlin/EvaluatorTest.kt]: Maybe use the contained `no_gpl_declared.kts` here to also ensure this embedded rule compiles?

File: evaluator/src/test/kotlin/EvaluatorTest.kt (hunk 1 above)
Comment (mnonnenmacher) [reply evaluator/src/test/kotlin/EvaluatorTest.kt]: Done.

File: evaluator/src/test/kotlin/EvaluatorTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply evaluator/src/test/kotlin/EvaluatorTest.kt]: Good idea, done.

**Pull request merged**